feat(iotda): add batchtask resource support (#4408)
huaweicloud/terraform-provider-huaweicloud: docs/resources/iotda_batchtask.md

  to view the HTTPS application access address. An example of the access address might be
  **9bc34xxxxx.st1.iotda-app.ap-southeast-1.myhuaweicloud.com**, then you need to configure the
  `provider` block as follows:

  ```hcl
  provider "huaweicloud" {
    endpoints = {
      iotda = "https://9bc34xxxxx.st1.iotda-app.ap-southeast-1.myhuaweicloud.com"
    }
  }
  ```

## Example Usage

```hcl
variable "task_name" {}
variable "task_type" {}
variable "targets_file" {}

resource "huaweicloud_iotda_batchtask" "test" { 
  name         = var.task_name
  type         = var.task_type
  targets_file = var.targets_file
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the IoTDA batch task resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

* `name` - (Required, String, ForceNew) Specifies the batch task name. The length does not exceed `128`, and only
  combinations of Chinese characters, letters, numbers, and underscores (_) are allowed.
  Changing this parameter will create a new resource.

> * `type` - (Required, String, ForceNew) Specifies the batch task type.  

Reviewer: ```suggestion
* `type` - (Required, String, ForceNew) Specifies the batch task type.  Changing this parameter will create a new resource.
```

  The valid values are as follows:
  + **createDevices**: Batch create devices task.
  + **updateDevices**: Batch update devices task.
  + **deleteDevices**: Batch deletion of devices task.
  + **freezeDevices**: Batch freeze devices task.
  + **unfreezeDevices**: Batch unfreeze devices task.

* `space_id` - (Optional, String, ForceNew) Specifies the resource space ID to which the batch task belongs.
  Changing this parameter will create a new resource.

* `targets` - (Optional, List, ForceNew) Specifies the target for executing batch task, fill in the device ID list here,
  and support a maximum of `30000` device IDs. This parameter is supported when the `type` is **deleteDevices**,

>   **freezeDevices**, and **unfreezeDevices**. Changing this parameter will create a new resource.

Reviewer: ```suggestion
  **freezeDevices**, or **unfreezeDevices**. Changing this parameter will create a new resource.
```
Check other parts.

* `targets_filter` - (Optional, Map, ForceNew) Specifies the batch task target filtering parameters.

>   (K, V) format, currently supported K includes group_ids (V fills in group ID array), e.g. **{group_ids = "xxxxxx"}**.

Reviewer: The description of this line of documentation is ambiguous. 
The documentation states `(V fills in group ID array` but the example given is a string.

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the IoTDA batch task resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

* `name` - (Required, String, ForceNew) Specifies the batch task name. The length does not exceed `128`, and only
  combinations of Chinese characters, letters, numbers, and underscores (_) are allowed.
  Changing this parameter will create a new resource.

* `type` - (Required, String, ForceNew) Specifies the batch task type.
  Changing this parameter will create a new resource.  
  The valid values are as follows:
  + **createDevices**: Batch create devices task.
  + **updateDevices**: Batch update devices task.
  + **deleteDevices**: Batch deletion of devices task.
  + **freezeDevices**: Batch freeze devices task.
  + **unfreezeDevices**: Batch unfreeze devices task.

* `space_id` - (Optional, String, ForceNew) Specifies the resource space ID to which the batch task belongs.
  Changing this parameter will create a new resource.

* `targets` - (Optional, List, ForceNew) Specifies the target for executing batch task, fill in the device ID list here,

>   and support a maximum of `30000` device IDs. This parameter is supported when the `type` is **deleteDevices**,

Reviewer 2: ```suggestion
* `targets` - (Optional, List, ForceNew) Specifies an array of target device IDs for executing the batch task, which can
  include up to 30,000 device IDs. This parameter is supported when the `type` is **deleteDevices**,
```

  **freezeDevices**, or **unfreezeDevices**. Changing this parameter will create a new resource.

* `targets_filter` - (Optional, List, ForceNew) Specifies the batch task target filtering parameters.
  Using this parameter, batch tasks will filter out devices that meet the criteria as targets.
  This parameter is supported when the `type` is **deleteDevices**, **freezeDevices**, or **unfreezeDevices**.
  Changing this parameter will create a new resource.

> The [targets_filter](#IoTDA_targets_filter) structure is documented below.

Reviewer: The spaces.

* `targets_file` - (Optional, String, ForceNew) Specifies the batch task file path to be used for creating batch task.
  Currently, only the **xlsx/xls** file format is supported, and the maximum number of lines in the file is `30000`.
  This parameter is supported when the `type` is **createDevices**, **updateDevices**, **deleteDevices**,
  **freezeDevices**, or **unfreezeDevices**. Changing this parameter will create a new resource.
  Please following [reference](https://support.huaweicloud.com/intl/en-us/usermanual-iothub/iot_01_0032.html),
  download the template file and fill it out.

> -> `targets`, `targets_filter` and `targets_file` cannot all be empty and can only be set to one.

Reviewer 2: ```suggestion
-> Exactly one of `targets`, `targets_filter` and `targets_file` should be specified.
```
